core/src/transaction.rs

```diff
@@ -74,141 +66,158 @@ impl rlp::Decodable for UnverifiedTransaction {
     }
 }
 
+impl SignedTransaction {
+    /// Append object with a signature into RLP stream
+    fn rlp_append_sealed_transaction(&self, s: &mut RlpStream) {
+        s.begin_list(6);
+        s.append(&self.unsigned.seq);
+        s.append(&self.unsigned.fee);
+        s.append(&self.unsigned.network_id);
+        s.append(&self.unsigned.action);
+        s.append(&self.sig);
+        s.append(&self.signer_public);
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct VerifiedTransaction(SignedTransaction);
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct UnverifiedTransaction(SignedTransaction);
+
+impl rlp::Encodable for VerifiedTransaction {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        self.0.rlp_append(s);
+    }
+}
+
+impl rlp::Decodable for VerifiedTransaction {
```

Comment:
I'm sorry that I recommended you to implement `rlp::Decodable` for `VerifiedTransaction` for the case of db backup. However it is inappropriate. `decode` is just in charge of converting serialized bytes into target structs. In case of db corruption, it seems ok to be panicked when recovering `MemPool` from the db. However, this general method may be used somewhere else then the `expect` and `panic` seem dangerous. I think it's better to elevate the panic behavior to higher level such as `recover_to_data` function.

Reply:
I modified it. Could you check the last commit?